sympy/functions/combinatorial/factorials.py

```diff
@@ -1057,3 +1058,93 @@ def _eval_is_nonnegative(self):
                 return True
             elif k.is_even is False:
                 return  False
+
+class multinomial(CombinatorialFunction):
+    r"""Implementation of the multinomial coefficient. They can be expressed as
+    a product of binomial coefficients or of factorials:
+
+    .. math:: \frac{(k_1+k_2+\cdots+k_m)!}{k_1!\, k_2! \cdots k_m!} = {k_1\choose k_1}{k_1+k_2\choose k_2}\cdots{k_1+k_2+\cdots+k_m\choose k_m}
+
+    The multinomial coefficients have a direct combinatorial interpretation,
+    as the number of ways of depositing n distinct objects into m distinct bins,
+    with k1 objects in the first bin, k2 objects in the second bin, and so on.
+
+    To expand the multinomial when `k` contains a symbol, use either
+    ``expand_func()`` or ``expand(func=True)``. The former will keep
+    the polynomial in factored form while the latter will expand the
+    polynomial itself. See examples for details.
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol, Rational, multinomial, expand_func, symbols
+    >>> n = Symbol('n', integer=True, positive=True)
+
+    >>> multinomial(15, 8)
+    490314
+
+    >>> multinomial(n, -1)
+    0
+
+    Comparing Coefficents generated with the multinomial function:
+
+    >>> a, b, c = symbols('a b c')
+    >>> ((a+b+c)**3).expand(func=True)
+    a**3 + 3*a**2*b + 3*a**2*c + 3*a*b**2 + 6*a*b*c + 3*a*c**2 + b**3 + 3*b**2*c + 3*b*c**2 + c**3
+
+    The Coefficient when of 2 1 0 corresponding to the second term is 3
+
+    >>> multinomial(2,1,0)
+    3
+
+    >>> multinomial(Rational(5, 4), 3)
+    gamma(21/4)/(6*gamma(9/4))
+
+    >>> multinomial(n, 3)
+    gamma(n + 4)/(6*gamma(n + 1))
+
+    >>> multinomial(n, 3).expand(func=True)
+    n**3/6 + n**2 + 11*n/6 + 1
+
+    >>> expand_func(multinomial(n, 3))
+    (n + 1)*(n + 2)*(n + 3)/6
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Multinomial_theorem#Multinomial_coefficients
+
+    .. [2] http://functions.wolfram.com/GammaBetaErf/Multinomial/02/0001/
+
+    """
+
+    @classmethod
+    def eval(cls, *k):
+        k = [sympify(x) for x in k]
+        numerator = Add(*k)
+        if all(x.is_Integer for x in k) and numerator > 0:
```

Comment:
I'm not sure the `numerator > 0` serves a purpose. Without the condition, any negative ki will return a result of 0 while all zeros will return 1 (as it does now).

Reply:
![This is the Formal Definition](http://functions.wolfram.com/GammaBetaErf/Multinomial/02/0001/MainEq1)   
This is the Formal Definition but it does give leeway to define the function without that condition. Adding it just makes it computationally efficient ... as in 
![This ](http://functions.wolfram.com/GammaBetaErf/Multinomial/02/0002/MainEq1)

Reply:
Instead of having the numerator checked over there ... I can add a condition before that ... such as 
```
if numerator < 0:   
  return 0
```
How does that sound ?

Reply:
I think it's more like
```python
if all(i.is_Integer for i in k):
    if any(i < 0 for i in k):
        return S.Zero
    return factorial answer
```